File: common/src/main/java/gaode/zhongjh/com/common/listener/VideoEditListener.java
package gaode.zhongjh.com.common.listener;

/**
 * 视频编辑的回调
 * @author zhongjh
 */
public interface VideoEditListener {

    /**
     * 完成
     */
    void onFinish();

    /**
     * 进度
     * @param progress 进度百分比
     * @param progressTime 进度时间
     */
    void onProgress(int progress, long progressTime);

    /**
     * 取消
     */
    void onCancel();

    /**
     * 异常
     * @param message 信息
     */
    void onError(String message);

}

File: common/src/main/java/com/zhongjh/common/listener/VideoEditListener.kt
package com.zhongjh.common.listener

/**
 * 视频编辑的回调
 * @author zhongjh
 * @date 2021/12/9
 */
interface VideoEditListener {

    /**
     * 完成
     */
    fun onFinish()

    /**
     * 进度
     * @param progress 进度百分比
     * @param progressTime 进度时间
     */
    fun onProgress(progress: Int, progressTime: Long);

    /**
     * 取消
     */
    fun onCancel();

    /**
     * 异常
     * @param message 信息
     */
    fun onError(message: String);

}